api/v1alpha2/linodefirewall_types.go

```diff
@@ -39,7 +38,12 @@ type LinodeFirewallSpec struct {
 	Enabled bool `json:"enabled,omitempty"`
 
 	// +optional
-	InboundRules []FirewallRule `json:"inboundRules,omitempty"`
+	InboundRules []FirewallRuleSpec `json:"inboundRules,omitempty"`
+
+	// InboundRuleRefs is a list of references to FirewallRules as an alternative to
+	// using InboundRules but can be used in conjunction with it
+	// +optional
+	InboundRuleRefs []*corev1.ObjectReference `json:"inboundRuleRefs,omitempty"`
```

Comment:
We might want to have some sort of validation that limits the maximum number of rules (across inbound and outbound rulesets) to 25: https://techdocs.akamai.com/linode-api/reference/post-firewalls

Reply:
We do that check [here](https://github.com/linode/cluster-api-provider-linode/blob/main/internal/controller/linodefirewall_controller_helpers.go#L461) but we could add some extra pre-validation on a webhook

Reply:
Sounds good! But it doesn't need to need to be a blocker here 🙅‍♀️